larpandoracontent/LArHelpers/LArFormattingHelper.h

```diff
@@ -0,0 +1,222 @@
+        DARK_GRAY = 90,
+        LIGHT_RED = 91,
+        LIGHT_GREEN = 92,
+        LIGHT_YELLOW = 93,
+        LIGHT_BLUE = 94,
+        LIGHT_MAGENTA = 95,
+        LIGHT_CYAN = 96,
+        WHITE = 97
+    };
+
+    //--------------------------------------------------------------------------------------------------------------------------------------
+
+    class Table 
+    {
+    public:
+
+        /**
+         * @brief  Table constructor
+         *
+         * @param  columnTitles the string columns titles use empty string for a separator column
+         * @param  precision the number of significant figures to display for number type table elements
+         */
+        Table(const std::vector<std::string> &columnTitles, const unsigned int &precision=3);
+
+        /**
+         * @brief  Add an element to the table into the next (non-separator) column
+         *
+         * @param  value the element to add to the table
+         * @param  style the style of the element
+         * @param  colour the colour of the element
+         */
+        template<class T>
+        void AddElement(const T &value, const Style &style=REGULAR, const Colour &colour=DEFAULT);
+
+        /**
+         * @brief  Print the table
+         */
+        void Print();
+    private:
+        const std::vector<std::string>  m_columnTitles;     ///< The vector of columns titles in the table
+        const unsigned int              m_precision;        ///< The number of significant figures to use when displaying number types
+        std::vector<std::string>        m_elements;         ///< The vector of flattened table elements
+        std::vector<std::string>        m_format;           ///< The formatting of each table element
+        std::vector<unsigned int>       m_widths;           ///< The widths of each column (in units of number of characters)
+        std::stringstream               m_stringstream;     ///< The stringstream to print objects to
+
+        /**
+         *  @brief  Check if the next table cell is in a separator column, if so add a blank element
+         */
+        void CheckForSeparatorColumn();
+
+        /**
+         *  @brief  If the supplied column is a separator (vertical rule)
+         *
+         *  @param  column the column index to check
+         *
+         *  @return true/false
+         */
+        bool IsSeparatorColumn(const unsigned int &column);
+    
+        /**
+         *  @brief  Update the width of the last column in which an element was added
+         */
+        void UpdateColumnWidth();
+
+        /**
+         *  @brief  Print the column titles
+         *
+         *  @param  widths a vector of the number of characters to include in each column 
+         */
+        void PrintColumnTitles();
+
+        /**
+         *  @brief  Print a horizontal line
+         *
+         *  @param  widths a vector of the number of characters to include in each column 
+         */
+        void PrintHorizontalLine();
```

Comment:
Check to see which member functions can be const. In class description, typically order by:
1. public daughter classes
2. public and const member functions (and public constructor, destructor)
3. public and non-const member functions
4. public member variables (often deprecated)
5. private daughter classes
6. private and const member functions
7. private and non-const member functions
8. private member variables

Reply:
Added const where possible, and modified order